server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java

```diff
@@ -35,29 +37,46 @@
  */
 public class RemoteStoreNodeAttribute {
 
-    public static final String REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX = "remote_store";
+    public static final List<String> REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX = List.of("remote_store", "remote_publication");
+    public static final String REMOTE_STORE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEY = "remote_store.state.repository";
     public static final String REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY = "remote_store.segment.repository";
     public static final String REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY = "remote_store.translog.repository";
-    public static final String REMOTE_STORE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEY = "remote_store.state.repository";
+    public static final String REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY = "remote_store.routing_table.repository";
+
+    public static final List<String> REMOTE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEYS = REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX.stream()
+        .map(prefix -> prefix + ".state.repository")
+        .collect(Collectors.toList());
+
+    public static final List<String> REMOTE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEYS = REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX.stream()
+        .map(prefix -> prefix + ".routing_table.repository")
+        .collect(Collectors.toList());
+    public static final List<String> REMOTE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEYS = REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX.stream()
+        .map(prefix -> prefix + ".segment.repository")
+        .collect(Collectors.toList());
+    public static final List<String> REMOTE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEYS = REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX.stream()
+        .map(prefix -> prefix + ".translog.repository")
+        .collect(Collectors.toList());
+
```

Comment:
why it has to apply to all repos. This work around with different prefix is mainly for clusters switching from remote store enabled to just remote publication enabled. Just apply to state + routing table

Reply:
The existing implementation to build Repository metdata does not always look-up for a specific attribute key from the node.  There are places in code which look up attributes based on regexp of prefix [1](https://github.com/rajiv-kv/OpenSearch/blob/bb45f0343553944627b18a3fd8fa0b79d0991b69/server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java#L174) [2](https://github.com/rajiv-kv/OpenSearch/blob/bb45f0343553944627b18a3fd8fa0b79d0991b69/server/src/main/java/org/opensearch/cluster/node/DiscoveryNode.java#L593).
To avoid handling combinations of prefix match, i have kept the prefix to be applied for all the configurable repositories.